### DISCUSSION TITLE
What is the way to prepend the file list ?

### DISCUSSION THREAD

**Discussion by MamaTierra** [ibhagwan/fzf-lua#1493] [Q&A]
Similarly to this issue https://github.com/ibhagwan/fzf-lua/issues/931 , I'm trying to add recent file on top of fzf_lua.files. I am currently able to print the selected file from custom list from fre:
```
local function fre_combined_fzf()
	local fre_output = vim.fn.systemlist("fre --sorted")
	print("fre_output:", vim.inspect(fre_output))

	fzf_lua.fzf_exec(fre_output, {
		actions = {
			["default"] = function(selected)
				local file = selected[1]
				print("exec:", file)
				vim.fn.system("fre add " .. file)
				require("fzf-lua.actions").file_edit_or_qf(file) -- does not work
			end,
		},
	})
end
```

But I'm struggling to 1) edit it and 2) merge it with fzf_lua.files.

Could you please give me an example to help me to achieve this ?

**Comment by ibhagwan** [comment]
See #1483

**Comment by ibhagwan** [comment]
> But I'm struggling to 1) edit it and 2) merge it with fzf_lua.files.

1. I can’t see the output of your vim.inspect so I can’t comment on that specifically but you need to edit the table entries yourself or create a new list of strings from it using string.format
2. Have you tried `fzf_lua.files({ cmd = 'fre --sorted' })`?

**Comment by MamaTierra** [comment]
Thank you for the quick answer (and for all the good work in this plugin btw).

1.
Here is the logs of the upper function:
```
 msg_show fre_output: { "lua/plugins/obsidian.lua", "init.lua", "lua/lazy.lua" }
 msg_show exec: lua/lazy.lua
 notify.error [Fzf-lua] fn_selected threw an error: ... /.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/path.lua:364: attempt to index local 'haystack' (a nil value)
stack traceback:
	...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/core.lua:242: in function '__index'
	...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/path.lua:364: in function 'lastIndexOf'
	...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/path.lua:369: in function 'stripBeforeLastOccurrenceOf'
	...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/path.lua:419: in function 'entry_to_file'
	...e/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/actions.lua:253: in function 'sel_to_qf'
	...e/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/actions.lua:306: in function 'file_edit_or_qf'
	/home/pierre/.config/nvim/lua/pierre/plugins/fzf.lua:25: in function 'action'
	...e/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/actions.lua:99: in function 'act'
	...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/core.lua:148: in function 'fn_selected'
	...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/core.lua:235: in function <...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/core.lua:234>
	[C]: in function 'xpcall'
	...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/core.lua:234: in function <...erre/.local/share/nvim/lazy/fzf-lua/lua/fzf-lua/core.lua:226>
```
The first part seem ok to me but I don't understand yet the error when passing the selected file in `file_edit_or_qf`

2.
`fzf_lua.files({ cmd = 'fre --sorted' })`  solved indeed the first step of what i'm trying to do: merge normal entries from fzf_lua.files + entries from `fre --sorted`.


I also took a look at https://github.com/ibhagwan/fzf-lua/discussions/1483 but my final cmd is just `fd --color=never --type f --hidden --follow --exclude .git` as if `vim.api.nvim_list_bufs()` was empty.

I'm very new to lua so i'm not able to understand this yet, I'll improve my lua skills and dig more into this.

**Reply by ibhagwan** [reply]
> The first part seem ok to me but I don't understand yet the error when passing the selected file in file_edit_or_qf

This means the entry you sent to the action is probably not in the expected formatting, you’ll need to do some old fashioned lua debugging to solve it :)

> I'm very new to lua so i'm not able to understand this yet, I'll improve my lua skills and dig more into this.

If you wish to customize neovim you’ll need to learn some lua, no way around it, fortunately it’s a very simple and easy to learn language.
